### DISCUSSION TITLE
Auto Fill in class structure from python?

### DISCUSSION THREAD

**Discussion by cessna85** [NationalSecurityAgency/ghidra#7988] [Q&A]
I've got a script that can recognize which functions are a __thiscall but I would like to also be able to fill in the class structure based on the newly added class funcs.

Is there a way to programmatically auto fill in a class structure?

**Comment by Javex** [comment]
I was also looking for this and found the UI function that performs this at https://github.com/NationalSecurityAgency/ghidra/blob/a0205ff9606d5309d9fbc085806ace91b6107d2b/Ghidra/Features/Decompiler/src/main/java/ghidra/app/plugin/core/decompile/actions/CreateStructureVariableAction.java#L82-L112

While searching I found that https://github.com/NationalSecurityAgency/ghidra/blob/a0205ff9606d5309d9fbc085806ace91b6107d2b/Ghidra/Features/Decompiler/ghidra_scripts/CreateStructure.java#L55 exists so there's already a Java script that can do this.

Note that you need to put your cursor on the first parameter to make this work for a "this" parameter. I've yet to figure out how to get that location from a function.
